### PR TITLE
Building wrapper outside the libikiwix  ,Refactoring wrapper to support libkiwix12 , CI support , CD deployment to maven.

### LINKED ISSUE #19: Refactoring wrapper to support latest libkiwix
Refactoring wrapper to support latest libkiwix

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
This PR contain the following.
a) Since we have removed the Java wrapper on 11.0.0 , so first building 11.0.0 with external java wrapper with building libzim and libkiwix so files.
b) on libkiwix 12, we have removed the Reader.cpp and Searcher.cpp so that requried lots of refracting on java wrapper side so that is being include.
c) Maven publishing gradle task to publish automatically .

### CODE REVIEW

**Review by kelson42 — CHANGES_REQUESTED** [review]
Add the github workflow to autopublish (look for example at `openzim/mwoffliner` or `openzim/sotoki` to look how it is done. Should be based on version tagging on `master`.

File: android-libkiwixbuild/app/src/main/cpp/kiwixsearcher.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  zim::SearchResultSet* cresult = &resultoutput;
+    if (cresult != nullptr) {
+    jclass resultclass
+        = env->FindClass("org/kiwix/kiwixlib/JNIKiwixSearcher$SearchResultSet");
+    jmethodID ctor = env->GetMethodID(
+        resultclass, "<init>", "(Lorg/kiwix/kiwixlib/JNIKiwixSearcher;JLorg/kiwix/kiwixlib/JNIKiwixSearcher;)V");
+    result = env->NewObject(resultclass, ctor, obj, reinterpret_cast<jlong>(new Handle<zim::SearchResultSet>(cresult)), obj);
+  }
+  return result;
+}
```
Comment (mgautierfr) [inline android-libkiwixbuild/app/src/main/cpp/kiwixsearcher.cpp:79]: Here you create a new search each time and you return a searchresultset.
But you must create the search only once and move the `getNextResult` function on the result set.
And I think we must not have a `getNextResult` we should have some kind of iterable structure (To define what it is in the java/kotlin word) and let the java/kotlin code iterate it.

**Review by mgautierfr — CHANGES_REQUESTED** [review]
I haven't take a real look on the compilation/publication side.
But the Reader/Searcher side yes, and I disagree with it.

The `kiwix::Reader` doesn't exist anymore. The wrapper must now wrap `zim::Archive` so the wrapper must now be called `zimarchive` not `kiwixreader`. This is the same thing for `kiwixsearcher`.

The API of libzim and libkiwix are fundamentally different, the wrapper should follow the API of zim object, not stick to the old libkiwix api. If we don't we will face a lot of issues and difficulties.

**Comment by rgaudin** [comment]
@MohitMaliFtechiz I will add redirects to nightlies so you don't have to find/guess date. I'll post back here once it's live.

**Comment by MohitMaliFtechiz** [comment]
> @MohitMaliFtechiz I will add redirects to nightlies so you don't have to find/guess date. I'll post back here once it's live.

Ok @rgaudin

**Comment by MohitMaliFtechiz** [comment]
> 

hi @rgaudin ,

We need dedicated redirect url for every android architecture for both libzim and libkiwix and also for Linux architecture as well so we need 10 urls.
a) Libzim :- arm , arm64-v8a, x86 , x86_64 , Linux
b) Libkiwix:- arm , arm64-v8a, x86 , x86_64 , Linux

**Comment by rgaudin** [comment]
I don't understand. This is what we currently have:

```
/nightly/libzim_macos-x86_64.tar.gz
/nightly/libzim_linux-x86_64.tar.gz
/nightly/libzim_android-arm.tar.gz
/nightly/libzim_android-arm64.tar.gz
/nightly/libzim_android-x86.tar.gz
/nightly/libzim_wasm-emscripten.tar.gz
/nightly/libzim_android-x86_64.tar.gz
```

```
/nightly/libkiwix_android-arm.tar.gz
/nightly/libkiwix_macos-x86_64.tar.gz
/nightly/libkiwix_android-x86.tar.gz
/nightly/libkiwix_android-x86_64.tar.gz
/nightly/libkiwix_linux-x86_64.tar.gz
/nightly/libkiwix_android-arm64.tar.gz
```

**Comment by MohitMaliFtechiz** [comment]
> I don't understand. This is what we currently have:
> 
> ```
> /nightly/libzim_macos-x86_64.tar.gz
> /nightly/libzim_linux-x86_64.tar.gz
> /nightly/libzim_android-arm.tar.gz
> /nightly/libzim_android-arm64.tar.gz
> /nightly/libzim_android-x86.tar.gz
> /nightly/libzim_wasm-emscripten.tar.gz
> /nightly/libzim_android-x86_64.tar.gz
> ```
> 
> ```
> /nightly/libkiwix_android-arm.tar.gz
> /nightly/libkiwix_macos-x86_64.tar.gz
> /nightly/libkiwix_android-x86.tar.gz
> /nightly/libkiwix_android-x86_64.tar.gz
> /nightly/libkiwix_linux-x86_64.tar.gz
> /nightly/libkiwix_android-arm64.tar.gz
> ```

ok understood this will work.

**Comment by kelson42** [comment]
@MohitMaliDeveloper What should we do with this PR

**Comment by MohitMaliFtechiz** [comment]
> @MohitMaliFtechiz @ 2 weeks have passed and still no answer?!

We already have this changes on main branch , i think we can close this PR.

**Pull request closed without merging**